examples/renderFooterInTable.js

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable no-console,func-names,react/no-multi-comp */
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Table from 'rc-table';
+import 'rc-table/assets/index.less';
+
+const columns = [
+  { title: 'Name', dataIndex: 'name', key: 'name', width: 100, fixed: true },
+  { title: 'Age', dataIndex: 'age', key: 'age', width: 100 },
+  { title: 'Height', dataIndex: 'height', key: 'height', width: 200 },
+  {
+    title: 'Operations', dataIndex: '', key: 'd', render() {
+      return <a href="#">Operations</a>;
+    },
+  },
+];
+
+const data = [
+  { name: 'Alice', age: 20, height: 158 },
+  { name: 'Bob', age: 27, height: 175 },
+  { name: 'Cathy', age: 19, height: 164 },
+];
+
+ReactDOM.render(
+  <div>
+    <h2>Render Footer in Table</h2>
+    <Table
+      columns={columns}
+      data={data}
+      style={{ width: 600 }}
+      scroll={{ x: 800, y: 100 }}
+      renderFooterInTable={true}
+      title={currentData => <div>Title: {currentData.length} items</div>}
+      footer={(currentData, columns, Row, Cell) => {
+        const cells = columns.map(col => {
+          const {key, dataIndex} = col;
+          let text;
+          switch (dataIndex) {
+            case 'name':
+              text = 'Summary';
+              break;
+            case 'age':
+              text = currentData.reduce((acc, record) => acc + record[dataIndex], 0);
+              text = 'Total Age: ' + text;
+              break;
+            case 'height':
+              text = currentData.reduce((acc, record) => acc + record[dataIndex], 0) / currentData.length;
+              text = 'Average Height: ' + text.toFixed(2);
+              break;
+          }
+
+          return <Cell key={key || dataIndex}>{text}</Cell>
+        })
+
+        return <Row>{cells}</Row>
```

Comment:
Is returning `<Row>` required?

Reply:
Not really. Only when you set `renderFooterInTable` to true should you return `<Row />`.

Reply:
I keep the original `footer` API untouched. Pass extra args to it, but only takes effect when you toggle `renderFooterInTable` on.